core/serialization/download_xlsx_serializer.py

```diff
@@ -0,0 +1,608 @@
+            places_metadata[outcome_data.programme.programme_id],
+            organisation_metadata[outcome_data.programme.programme_id],
+        ).to_dict()
+        for outcome_data in programme_outcomes
+    ]
+
+    # Project level data
+    project_ids = ids(projects)
+    project_details = get_project_details(projects, organisation_metadata, places_metadata)
+    project_progresses = get_project_progresses(project_ids, places_metadata, organisation_metadata)
+    funding = get_funding(project_ids, places_metadata, organisation_metadata)
+    funding_comments = get_funding_comments(project_ids, places_metadata, organisation_metadata)
+    private_investments = get_private_investments(project_ids, places_metadata, organisation_metadata)
+    outputs_ref = [OutputDimSerializer(output_dim).to_dict() for output_dim in ents.OutputDim.query.all()]
+    output_data = get_outputs(project_ids, places_metadata, organisation_metadata)
+    outcome_ref = [OutcomeDimSerializer(outcome_dim).to_dict() for outcome_dim in ents.OutcomeDim.query.all()]
+    project_outcomes = [
+        OutcomeDataSerializer(
+            outcome_data,
+            places_metadata[outcome_data.programme.programme_id],
+            organisation_metadata[outcome_data.programme.programme_id],
+        ).to_dict()
+        for outcome_data in project_outcomes
+    ]
+
+    risks = get_risks(
+        programme_ids, project_ids, places_metadata, organisation_metadata
+    )  # risks are combination of programme and project risks
+    outcomes = [*programme_outcomes, *project_outcomes]  # outcomes are combination of programme and project outcomes
+
+    # Provides sheet order for when outputted in Excel format.
+    download_data = {
+        "PlaceDetails": place_details,
+        "ProjectDetails": project_details,
+        "OrganisationRef": organisation_refs,
+        "ProgrammeRef": programme_refs,
+        "ProgrammeProgress": programme_progresses,
+        "ProjectProgress": project_progresses,
+        "FundingQuestions": funding_questions,
+        "Funding": funding,
+        "FundingComments": funding_comments,
+        "PrivateInvestments": private_investments,
+        "OutputsRef": outputs_ref,
+        "OutputData": output_data,
+        "OutcomeRef": outcome_ref,
+        "OutcomeData": outcomes,
+        "RiskRegister": risks,
+    }
+    return download_data
+
+
+def org_metadata_programme_id(programme_ids):
+    organisations = {}
+
+    organisation_names = get_programme_child_with_natural_keys_query(ents.PlaceDetail, programme_ids).filter_by(
+        question="Grant Recipient:"
+    )
+
+    for org in organisation_names:
+        organisations[org.programme.programme_id] = org.answer
+    return organisations
+
+
+def places_metadata_programme_id(programme_ids):
+    places = {}
+    place_name = get_programme_child_with_natural_keys_query(ents.PlaceDetail, programme_ids).filter_by(
+        question="Please select your place name"
+    )
```

Comment:
I think the filter for a particular question should be as part of the previous query for efficiency. Might be best to just make a bespoke query for this function rather than reusing the existing function in this case